templates/vue/src/App.vue

```diff
@@ -0,0 +1,165 @@
+                  :networkId="networkId"
+                  :msg="'called method: setGreeting'"
+                  :visible="false"
+                  />
+  </template>
+
+  <template v-else>
+    <main>
+      <h1>Welcome to NEAR!</h1>
+      <p>
+        To make use of the NEAR blockchain, you need to sign in. The button
+        below will sign you in using NEAR Wallet.
+      </p>
+      <p>
+        By default, when your app runs in "development" mode, it connects
+        to a test network ("testnet") wallet. This works just like the main
+        network ("mainnet") wallet, but the NEAR Tokens on testnet aren't
+        convertible to other currencies - they're just for testing!
+      </p>
+      <p>
+        Go ahead and click the button below to try it out:
+      </p>
+      <p style="text-align:center; margin-top:2.5em">
+        <button v-on:click="login">Sign in</button>
+      </p>
+    </main>
+  </template>
+</div>
+</template>
+
+<script>
+
+import getConfig from './config'
+const nearConfig = getConfig(process.env.NODE_ENV || 'development')
+
+console.log(`networkId:${nearConfig.networkId} CONTRACT_NAME:${nearConfig.contractName}`);
+
+import { login, logout, onSubmit } from './utils'  
+import './global.css'
+
+import Notification from './components/Notification.vue'
+
+export default {
+  created() {
+    document.title = "near-blank-project"
+    if (this.signedIn) {
+        this.retrieveSavedGreeting()
+    }
+  },
+  name: 'App',
+  components: {
+    Notification
+  },
+  data: function () {
+    return {
+      savedGreeting: "Hello",
+      newGreeting: "Hello",
+      notificationVisible: false,
+      networkId: nearConfig.networkId,
+    }
+  },
+  computed: {
+    signedIn() { return window.walletConnection.isSignedIn() },
+    accountId() { return window.accountId }
+  },
+
+  methods: {
+
+    login() {
+      console.log("calling utils.login")
+      login()
+      this.retrieveSavedGreeting()
+    },
+
+  logout: logout,
+
+  retrieveSavedGreeting: () => {
+    //retrieve greeting
+    window.contract.getGreeting({ accountId: window.accountId })
+      .then(greetingFromContract => {
+        this.savedGreeting = greetingFromContract
+        this.newGreeting = greetingFromContract
+    })
+  },
+
+  saveGreeting: async function (event) {
+    this.updating = true;
+    // fire frontend-agnostic submit behavior, including data persistence
+    // look in utils.js to see how this updates data on-chain!
+    await onSubmit(event) //gets data directly from HTML elements
+
+    // update upper `greeting` message with persisted value
+    this.savedGreeting = this.newGreeting;
+    this.updating = false
+
+    this.notificationVisible = true; //show new notification
+
+    // remove Notification again after css animation completes
+    // this allows it to be shown again next time the form is submitted
+    setTimeout(() => {
+      this.notificationVisible = false
+    }, 11000)
+    }
+  }
```

Comment:
Is this a minor indentation error or an extra closing bracket?

Reply:
indentation error, fixed, now in src/components/SignedIn.vue